### ISSUE TITLE
Views: The view `index.ejs` should be rendered by `indexRouter` instead of `app.js`

### ISSUE THREAD

**Issue by RushilJalal** [TheOdinProject/curriculum#29188]
### Checks

- [X] This is not a duplicate of an existing issue (please have a look through our [open issues list](https://github.com/TheOdinProject/curriculum/issues) to make sure)
- [X] I have thoroughly read and understand [The Odin Project Contributing Guide](https://github.com/TheOdinProject/.github/blob/main/CONTRIBUTING.md)
- [X] Would you like to work on this issue?

### Describe your suggestion

Here's the screenshot for understanding:

![image](https://github.com/user-attachments/assets/e0a5da97-fa30-4a79-ba77-c771bacadcbb)

I believe if we're trying to render a view for `index`, it should be done in the `indexRouter` file instead of `app.js`, which can lead to cluttered code if all the views for all the routes are rendered in the main file `app.js`.

I am just a beginner so correct me if I'm wrong. Thanks!

### Path

Node / JS

### Lesson Url

https://www.theodinproject.com/lessons/nodejs-views

### (Optional) Discord Name

willyWonka

### (Optional) Additional Comments

I would love to contribute if possible!

**Comment by JE-Richards** [comment]
I was about to raise this. I hadn't realised `indexRouter.js` was suddenly being ignored in favour of directly controlling the route inside `app.js` and so it caused me a lot of confusion further into the lesson when reusable templates are introduced.

To be precise, there is a section where we create a `navbar.ejs` for rendering inside `index.ejs`, and the navbar needs to be passed in some `links` which we define in `app.js` for use across components. That specific section provides the following code:
```node
// app.js
const links = [
  { href: "/", text: "Home" },
  { href: "about", text: "About" },
];

app.get("/", (req, res) => {
  res.render("index", { links: links });
});
```
which only covers how to pass the links directly to the route within `app.js` and not how to pass it to the router created in `indexRouter.js`. 

I found a workaround, which is to refactor `indexRouter.js` into
```node
const { Router } = require('express');

const indexRouter = (links) => {
  const router = Router();

  router.get('/', (req, res) => res.render('index', { links: links }));

  return router;
};

module.exports = indexRouter;
```

and then in `app.js` call it like:
```node
app.use('/', indexRouter(links));
```

But as I'm still learning, I'm not sure if doing it this way is good practice or not, or whether it will conflict with anything covered in lessons further into the course.

**Comment by RushilJalal** [comment]
This issue is actually important and should not be marked as stale. TOP maintainers kindly look into this.

**Comment by CouchofTomato** [comment]
@TheOdinProject/react Can someone please take a look at this.

**Comment by KevinMulhern** [comment]
@01zulfi do you have any thoughts on this one?

**Comment by 01zulfi** [comment]
I don't think this is an issue. Having the route in the `app.js` makes the lesson easier to follow. We could add this sentence in the lesson:

> Ideally we would define the route in a separate router file, but for the purposes of this lesson we'll be defining it in `app.js`

**Comment by KevinMulhern** [comment]
@RushilJalal, sorry for the long delay, I appreciate it's been 7 months 😬. But do you still want to work on this one?

**Comment by RushilJalal** [comment]
@KevinMulhern  I would like to work on the issue.
Just to be clear, I have to add the sentence as mentioned by @01zulfi ?

**Comment by KevinMulhern** [comment]
Yep, that’s right - just the sentence should do it. Thanks for picking this up @RushilJalal ! 🙌
